src/us/kbase/workspace/database/exceptions/ErrorOr.java

```diff
@@ -0,0 +1,96 @@
+package us.kbase.workspace.database.exceptions;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+/** 
+ * A class that represents an object or an error that occurred attempting to create or retrieve
+ * the object.
+ */
+public class ErrorOr<T> {
+
+	// TODO ERROR may want to support including custom error fields? Different errors = different fields
+	// TODO LOMBOK
+	// TODO CODE return this from bulk methods rather than object / null 
+	
+	private final ErrorType error;
```

Comment:
I can see that coupling `ErrorType` with `ErrorTo` here might be pragmatic... however, I could see `ErrorOr` becoming something we use more broadly, in which case we'd want the error type to be a second generic.

Reply:
That could make sense, but in the case of the workspace not necessary, so I'm pulling a YAGNI here.

Reply:
Yes, we are saying the same thing in this case.